### PR TITLE
feat: ios scenario numbers

### LINKED ISSUE #1196: iOS | Add --scenario-numbers
**Author the user story for this feature**

As an iOS developer, I want to control scenario numbers so I can control which scenarios run.

**Describe the solution you'd like**

> --scenario-numbers=int,[int,…]
> A list of game-loop scenario numbers which will be run as part of the test (default: scenario 1). A maximum of 1024 scenarios may be specified in one test matrix, but the maximum number may also be limited by the overall test --timeout setting. This flag is only valid when --type=game-loop is also set.
> 
> https://cloud.google.com/sdk/gcloud/reference/alpha/firebase/test/ios/run#--scenario-numbers

### PR DESCRIPTION
**Pull request by Sloox**
Fixes #1196 

## Checklist

- [x] Documented
- [x] Unit tested

### CODE REVIEW

File: test_runner/src/main/kotlin/ftl/args/ValidateIosArgs.kt [hunk 1]
```diff
@@ -13,6 +13,13 @@ fun IosArgs.validate() = apply {
     assertTestFiles()
     checkResultsDirUnique()
     assertAdditionalIpas()
+    assertGameloop()
+}
+
+fun IosArgs.assertGameloop() {
+    /*if (scenarioNumbers.isNotEmpty() && (type == null || type != Type.GAMELOOP))
+        throw FlankConfigurationError("Scenario numbers defined but Type is not Game-loop.")*/
```
Comment (piotradamczyk5) [inline test_runner/src/main/kotlin/ftl/args/ValidateIosArgs.kt:21]: if this commented code needed?

Comment (Sloox) [reply test_runner/src/main/kotlin/ftl/args/ValidateIosArgs.kt:21]: this is a draft pull request. changes will come.

File: test_runner/src/main/kotlin/ftl/args/ValidateIosArgs.kt [hunk 2]
```diff
@@ -15,6 +15,13 @@ fun IosArgs.validate() = apply {
     checkResultsDirUnique()
     assertAdditionalIpas()
     validType()
+    assertGameloop()
+}
+
+fun IosArgs.assertGameloop() {
+    if (scenarioNumbers.isNotEmpty() && (type == null || type != Type.GAMELOOP))
```
Comment (piotradamczyk5) [inline test_runner/src/main/kotlin/ftl/args/ValidateIosArgs.kt:22]:
```suggestion
    if (scenarioNumbers.isNotEmpty() && (type != Type.GAMELOOP))
```

File: test_runner/src/main/kotlin/ftl/args/ValidateIosArgs.kt [hunk 3]
```diff
@@ -15,6 +15,13 @@ fun IosArgs.validate() = apply {
     checkResultsDirUnique()
     assertAdditionalIpas()
     validType()
+    assertGameloop()
+}
+
+fun IosArgs.assertGameloop() {
```
Comment (piotradamczyk5) [inline test_runner/src/main/kotlin/ftl/args/ValidateIosArgs.kt:21]: you could check also if `scenariosNumber.size` is below `1024`

File: test_runner/src/main/kotlin/ftl/args/ValidateIosArgs.kt (hunk 2 above)
Comment (Sloox) [reply test_runner/src/main/kotlin/ftl/args/ValidateIosArgs.kt:22]: 👍

File: test_runner/src/main/kotlin/ftl/args/ValidateIosArgs.kt (hunk 3 above)
Comment (Sloox) [reply test_runner/src/main/kotlin/ftl/args/ValidateIosArgs.kt:21]: 👍

**Review by piotradamczyk5 — APPROVED** [review]

**Review by pawelpasterz — APPROVED** [review]

**Pull request merged**